cpp/core/memory/MemoryAllocator.cc

```diff
@@ -166,20 +164,22 @@ bool StdMemoryAllocator::free(void* p, int64_t size) {
   return true;
 }
 
-bool StdMemoryAllocator::reserveBytes(int64_t size) {
-  bytes_ += size;
-  return true;
-}
-
-bool StdMemoryAllocator::unreserveBytes(int64_t size) {
-  bytes_ -= size;
-  return true;
-}
-
 int64_t StdMemoryAllocator::getBytes() const {
   return bytes_;
 }
 
+class NoopAllocationListener : public gluten::AllocationListener {
```

Comment:
There is a thinking that we should only derive from abstract classes in cpp programming.
You can get some information from google, and then make a decision. whatever you do, it will be ok for me.

Reply:
Wasn't `gluten::AllocationListener` already an abstract class? The code is

```cpp
class AllocationListener {
 public:
  static AllocationListener* noop();

  virtual ~AllocationListener() = default;

  // Value of diff can be either positive or negative
  virtual void allocationChanged(int64_t diff) = 0;

 protected:
  AllocationListener() = default;
};
```

Reply:
OK